git-pr-chain.py

```diff
@@ -700,6 +715,37 @@ def cmd_merge(args):
     # evil global caches, which are now out of date because we've merged our
     # PR!
 
+def cmd_new_pr(args):
+    # Use this command to mark the top commit as a new PR in the chain. It will
+    # generate the PR branch automatically
+    def generate_pr_chain_name(commit_title) -> str:
+        downcased = commit_title.lower()
+        # Change any non-words to _
+        only_word_char = re.sub(r"[^\w]", "_", downcased)
+        # Make _ compact: ____ -> _
+        truncate_continuous_underscore = re.sub(r"_+", "_", only_word_char)
+        # Strip _ away from the head and tail: ___xx_xx___ -> xx_xx
+        strip_start_end_underscore = re.sub(r"^_*|_*$", "", truncate_continuous_underscore)
+        # Only keep 40 characters
+        return strip_start_end_underscore[0:40]
```

Comment:
I'm wondering if we should add a nonce to this (probably sufficient to generate 64 random bits) so that branch names don't collide.  The first line of a commit isn't necessarily unique...

Reply:
This is the only thing that's keeping me from approving this as-is.